qcodes/instrument_drivers/rigol/DS1074Z.py

```diff
@@ -0,0 +1,207 @@
+
+        self.add_parameter('waveform_yorigin',
+                           get_cmd='WAVeform:YORigin?',
+                           unit='V',
+                           get_parser=float
+                           )
+
+        self.add_parameter('waveform_yincrem',
+                           get_cmd=':WAVeform:YINCrement?',
+                           unit='V',
+                           get_parser=float
+                           )
+
+        self.add_parameter('waveform_yref',
+                           get_cmd=':WAVeform:YREFerence?',
+                           unit='V',
+                           get_parser=float
+                           )
+
+        self.add_parameter('trigger_mode',
+                           get_cmd=':TRIGger:MODE?',
+                           set_cmd=':TRIGger:MODE {}',
+                           unit='V',
+                           vals=Enum('edge', 'EDGE',
+                                     'pulse', 'PULS', 'PULSE',
+                                     'video', 'VID', 'video',
+                                     'pattern', 'PATT', 'PATTERN'
+                                     ),
+                           get_parser=str
+                           )
+
+        # trigger source
+        self.add_parameter('trigger_level',
+                           unit='V',
+                           get_cmd=self._get_trigger_level,
+                           set_cmd=self._set_trigger_level,
+                           vals=Numbers()
+                           )
+
+        self.add_parameter('trigger_edge_source',
+                           label='Source channel for the edge trigger',
+                           get_cmd=':TRIGger:EDGE:SOURce?',
+                           set_cmd=':TRIGger:EDGE:SOURce {}',
+                           val_mapping={'ch1': 'CHAN1',
+                                        'ch2': 'CHAN2',
+                                        'ch3': 'CHAN3',
+                                        'ch4': 'CHAN4'
+                                        }
+                           )
+
+        self.add_parameter('trigger_edge_slope',
+                           label='Slope of the edge trigger',
+                           get_cmd=':TRIGger:EDGE:SLOPe?',
+                           set_cmd=':TRIGger:EDGE:SLOPe {}',
+                           vals=Enum('positive', 'negative', 'neither')
+                           )
+
+        self.add_parameter('data_source',
+                           label='Waveform Data source',
+                           get_cmd=':WAVeform:SOURce?',
+                           set_cmd=':WAVeform:SOURce {}',
+                           val_mapping={'ch1': 'CHAN1',
+                                        'ch2': 'CHAN2',
+                                        'ch3': 'CHAN3',
+                                        'ch4': 'CHAN4'
+                                        }
+                           )
+
+        self.add_parameter('time_axis',
```

Comment:
please set the snapshot_value to False for this parameter

Reply:
done